src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -194,6 +199,29 @@ protected String getLanguagePlaceholderText(@NotNull ASTNode node, @NotNull Text
         return null;
     }
 
+    private static boolean isCollapsedByDefault(@NotNull FoldingRange foldingRange) {
+        String foldingRangeKind = foldingRange.getKind();
+        if (foldingRangeKind != null) {
+            CodeFoldingSettings codeFoldingSettings = CodeFoldingSettings.getInstance();
+
+            // Imports
+            if (codeFoldingSettings.COLLAPSE_IMPORTS &&

```

Comment:
If the IDE is configured to collapse imports by default and this folding region's `kind` is `imports`, it should be collapsed by default.